src/test/java/pages/OrderConfirmationPage.java

```diff
@@ -0,0 +1,213 @@
+import utility.DateHandler;
+
+import java.util.Date;
+import java.util.List;
+
+public class OrderConfirmationPage {
+    private final WebDriver driver;
+    private final Actions action;
+
+    public OrderConfirmationPage(WebDriver driver)  {
+        PageFactory.initElements(driver, this);
+        this.driver = driver;
+        action = new Actions(driver);
+    }
+
+    private Date desiredDate;
+    private boolean desiredDateIsFuture, presentYear, presentMonth;
+
+    public void setDesiredDate(Date desiredDate)   {
+        this.desiredDate = desiredDate;
+        Date currentDate = new Date();
+        if(desiredDate.after(currentDate))   {
+            desiredDateIsFuture = true;
+        }
+        if (desiredDate.getYear() == currentDate.getYear())   {
+            presentYear = true;
+        }
+        if (desiredDate.getMonth() == currentDate.getMonth())   {
+            presentMonth = true;
+        }
+    }
+
+    @FindBy(xpath = "//label[contains(., 'Date picker')]/../div[contains(@class, 'Input')]")
+    private WebElement datePickerField;
+
+    @FindBy(xpath = "//label[contains(., 'Time picker')]/../div[contains(@class, 'Input')]")
+    private WebElement timePickerField;
+
+    @FindBy(xpath = "//div[contains(@class, 'CalendarHeader')]//button[1]")
+    private WebElement calendarLeftBtn;
+
+    @FindBy(xpath = "//div[contains(@class, 'CalendarHeader')]//button[2]")
+    private WebElement calendarRightBtn;
+
+    @FindBy(xpath = "//div[contains(@class, 'CalendarHeader')]//div")
+    private WebElement dateCurrentYearMonth;
+
+    @FindBy(xpath = "//button[.//span[text()='OK']]")
+    private WebElement calendarClockOkBtn;
+
+    @FindBy(xpath = "//*[text() = 'AM']")
+    private WebElement amBtn;
+
+    @FindBy(xpath = "//*[text() = 'PM']")
+    private WebElement pmBtn;
+
+    @FindBy(xpath = "//div[contains(@class, 'squareMask')]")
+    private WebElement clockTimePicker;
+
+    @FindBy(xpath = "//table//tbody//tr")
+    private List<WebElement> itemsInOrder;
+
+    @FindBy(xpath = "//table//*[text()='Total:']//..//td[3]")
+    private WebElement totalOrderPrice;
+
+    public void datePickerAccess()  {
+        datePickerField.click();
+    }
+
+    public void timePickerAccess()  {
+        timePickerField.click();
+    }
+
+    public void calendarAcceptChanges() {
+        calendarClockOkBtn.click();
+    }
+
+    public void setYearCalendar()  {
+        String date;
+        int year;
+        int desiredDateRealYear = getDesiredDateRealYear(desiredDate.getYear());
+
+        if(!presentYear)    {
+            if(!desiredDateIsFuture)  {
+                do
+                {
+                    calendarLeftBtn.click();
+                    date = dateCurrentYearMonth.getAttribute("innerText");
+                    year = Integer.parseInt(date.replaceAll("\\D", ""));
+                }   while (year != desiredDateRealYear);
+            }
+
+            if (desiredDateIsFuture) {
```

Comment:
This `if` statement is redundant, we can put the body to `else` block of previous `if`

Reply:
Actually i don't like 'else' statement, without it we are working with exactly desired condition